Remotion/Data/DomainObjects/Mapping/Builder/TypeDefinitionBuilderNode.cs

```diff
@@ -54,7 +54,7 @@ public void BeginBuildTypeDefinition (IMappingObjectFactory mappingObjectFactory
     /// <summary>
     /// Creates a <see cref="TypeDefinition"/> for this node and links it to its parents.
     /// </summary>
-    protected abstract void BuildTypeDefinition (IMappingObjectFactory mappingObjectFactory);
+    internal abstract void BuildTypeDefinition (IMappingObjectFactory mappingObjectFactory);
```

Comment:
We should use internal only for testing completely internal types. making protected members internal means we're breaking encapsulation because we no longer declare this api as only-inside-this-type(hierarchy). 

For testing protected apis there's always the option of deriving the type in the tests as we do already and provide a public method for testing via the testable-type.

Reply:
The internal here is necessary because `ClassDefinitionBuilderNode` needs to call `BuildTypeDefinition` on the `InterfaceDefinitionBuilderNode`. Making it public would cause problems as we now have a `BuildTypeDefinition` method in addition to `Start/EndBuildTypeDefinition`

Reply:
I think you had the idea of introducing another non-virtual method that calls the virtual method? But with that we still have the same problem don't we? The new method would have to be public for this to work.

In this case I would leave the code as is. Making the `BuildTypeDefinition` method public or adding another public method that essential just redirects to this method harms the usability more than it improves it.